docs/further_useful_features.rst

```diff
@@ -0,0 +1,459 @@
+underlying processes differ and creation of the :class:`~kartothek.core.index.PartitionIndex`
+is merely a side-effect of using ``partition_on``.
+
+
+Updating an existing dataset
+============================
+
+It's possible to update datasets by adding new physical partitions to them, ``kartothek``
+provides update functions that generally have the prefix `update_dataset` in their names.
+For example, :func:`~kartothek.io.eager.update_dataset_from_dataframes` is the update
+function for the ``eager`` backend.
+
+To see updating in action, let's set up some data and a storage location first and store
+the data there with ``kartothek``:
+
+.. ipython:: python
+
+    dm = store_dataframes_as_dataset(store_factory, "a_unique_dataset_identifier", df)
+    sorted(dm.partitions.keys())
+
+
+Now, we create ``another_df`` with the same schema as our intial dataframe
+``df`` and update it using the ``eager`` backend by calling :func:`~kartothek.io.eager.update_dataset_from_dataframes`:
+
+.. ipython:: python
+
+    from kartothek.io.eager import update_dataset_from_dataframes
+
+    another_df = pd.DataFrame(
+        {
+            "A": 5.0,
+            "B": pd.Timestamp("20110102"),
+            "C": pd.Series(2, index=list(range(4)), dtype="float32"),
+            "D": np.array([6] * 4, dtype="int32"),
+            "E": pd.Categorical(["prod", "dev", "prod", "dev"]),
+            "F": "bar",
+        }
+    )
+
+    dm = update_dataset_from_dataframes(
+        [another_df], store=store_factory, dataset_uuid="a_unique_dataset_identifier"
+    )
+    sorted(dm.partitions.keys())
+
+
+Looking at ``dm.partitions``, we can see that another partition has
+been added.
+
+If we read the data again, we can see that the ``another_df`` has been appended to the
+previous contents.
+
+.. ipython:: python
+
+    from kartothek.io.eager import read_table
+
+    updated_df = read_table("a_unique_dataset_identifier", store_factory, table="table")
+    updated_df
+
+
+The way dataset updates work is that new partitions are added to a dataset
+as long as they have the same tables as the existing partitions. A `different`
+table **cannot** be introduced into an existing dataset with an update.
+
+To illustrate this point better, let's first create a dataset with two tables:
+
+.. ipython:: python
+
+    df2 = pd.DataFrame(
+        {
+            "G": "foo",
+            "H": pd.Categorical(["test", "train", "test", "train"]),
+            "I": np.array([9] * 4, dtype="int32"),
+            "J": pd.Series(3, index=list(range(4)), dtype="float32"),
+            "K": pd.Timestamp("20190604"),
+            "L": 2.0,
+        }
+    )
+    df2
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "another_unique_dataset_identifier",
+        dfs=[{"data": {"table1": df, "table2": df2}}],
+    )
+    dm.tables
+    sorted(dm.partitions.keys())
+
+
+.. admonition:: Filenames
```

Comment:
This is rather what we refer to as partition labels. Do not refer to the explicit file names. People should trust and rely on kartothek to manage the files. The mentioning of the uniqueness of the labels is nice, though